examples/opensora_hpcai/opensora/models/layers/blocks.py

```diff
@@ -184,10 +185,7 @@ def construct(self, x, cond, mask=None):
         x = ops.reshape(x, (B, N, -1))
 
         # 4. output projection
```

Comment:
why change this?

Reply:
When using Flash Attention, the attention `dtype` changes to bf16, including `self.proj` and `self.proj_drop`. These should be cast back if the network uses a different data type. Otherwise training crash due to different data types.

Reply:
Then the type conversion should be put after FA instead of the block output?

Reply:
`self.proj` and `self.proj_drop` are also set to `attn_dtype`, so need to cast at the end of the `SelfAttention` and `MultiHeadCrossAttention` blocks.